app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedFragmentTest.kt

```diff
@@ -0,0 +1,894 @@
+    launch<MarkChaptersCompletedTestActivity>(
+      createMarkChaptersCompletedTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 1)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.mark_chapters_completed_recycler_view,
+          position = 1,
+          targetViewId = R.id.mark_chapters_completed_chapter_check_box
+        )
+      ).check(matches(not(isChecked())))
+      scrollToPosition(position = 2)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.mark_chapters_completed_recycler_view,
+          position = 2,
+          targetViewId = R.id.mark_chapters_completed_chapter_check_box
+        )
+      ).check(matches(not(isEnabled())))
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedFragment_configChange_firstStory_firstChapterUnchecked_secondChapterIsDisabled() { // ktlint-disable max-line-length
+    launch<MarkChaptersCompletedTestActivity>(
+      createMarkChaptersCompletedTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      scrollToPosition(position = 1)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.mark_chapters_completed_recycler_view,
+          position = 1,
+          targetViewId = R.id.mark_chapters_completed_chapter_check_box
+        )
+      ).check(matches(not(isChecked())))
+      scrollToPosition(position = 2)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.mark_chapters_completed_recycler_view,
+          position = 2,
+          targetViewId = R.id.mark_chapters_completed_chapter_check_box
+        )
+      ).check(matches(not(isEnabled())))
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedFragment_firstStory_selectChaptersInOrder_unselectFirstChapter_secondChapterIsDisabled() { // ktlint-disable max-line-length
+    launch<MarkChaptersCompletedTestActivity>(
+      createMarkChaptersCompletedTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 1)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 1)
+      scrollToPosition(position = 2)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 2)
+      scrollToPosition(position = 1)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 1)
+      scrollToPosition(position = 2)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.mark_chapters_completed_recycler_view,
+          position = 2,
+          targetViewId = R.id.mark_chapters_completed_chapter_check_box
+        )
+      ).check(matches(not(isEnabled())))
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedFragment_configChange_firstStory_selectChaptersInOrder_unselectFirstChapter_secondChapterIsDisabled() { // ktlint-disable max-line-length
```

Comment:
ditto as above, the `configChange` should take place after the action and verify that the intended outcome persists.

Reply:
Done.

Reply:
Done.